content/token22-metadata.md

```diff
@@ -0,0 +1,1083 @@
+---
+title: Metadata and Metadata Pointer Extension
+objectives:
+- Explain how the metadata pointers and metadata extensions work on Token Extension Program Mints
+- Create an NFT with metadata embedded in the mint account itself
+- Create an NFT with the metadata pointer extension
+---
+
+# Summary
+- The `metadata pointer` extension associates a public key directly to a metadata account, whether it's stored internally or via an external metadata account like Metaplex. This setup simplifies the association of tokens with their metadata.
+- The `metadata` mint extension allows embedding of metadata directly into mint accounts through the Token Extension Program, typically utilized alongside a self-referencing metadata pointer. This facilitates embedding comprehensive token information at the minting stage.
+- These extensions enhance the interoperability of tokens across different applications and platforms by standardizing how metadata is associated and accessed.
+- Directly embedding or pointing to metadata can streamline transactions and interactions by reducing the need for additional lookups or external calls.
+
+# Overview
+
+The Token Extension Program streamlines metadata on Solana. Without the Token Extension Program, developers store metadata in metadata accounts using a metadata on-chain program; mainly `Metaplex`. However, this has some drawbacks. For example the mint account to which the metadata is "attached" has no awareness of the metadata account. To determine if an account has metadata, we have to PDA the mint and the `Metaplex` program together and query the network to see if a Metadata account exists. Additionally, to create and update this metadata you have to use a secondary program (i.e. `Metaplex`). These processes introduces vender lock and increased complexity. Token Extension Programs's Metadata extensions fix this by introducing two extensions:
+
+- `metadata-pointer` extension: Adds two simple fields in the mint account itself: a publicKey pointer to the account that holds the metadata for the token following the [Token-Metadata Interface](https://github.com/solana-labs/solana-program-library/tree/master/token-metadata/interface), and the authority to update this pointer.
+- `metadata` extension: Adds the fields described in the [Token-Metadata Interface](https://github.com/solana-labs/solana-program-library/tree/master/token-metadata/) which allows us to store the metadata in the mint itself.
+
+Note: The `metadata` extension is usually used in conjunction with the `metadata-pointer` extension which points back to the mint itself.
```

Comment:
So: `metadata` allows us to store metadata directly in the token mint. Since we don't have a separate metadata account, what is being pointed to with metadata pointer?

Reply:
the pointer in this case will point to the mint itself

I think that should streamline the process of getting the metadata (fetch mint => get the pointer => get the metadata from the pointer) 

in this case, we don't have to always check if the mint has the metadata embedded inside it or not

this is what I understood from it, not sure though.

it might also be for a security reason, so we lock in the metadata for this mint to the mint itself rather than keeping it open

Reply:
Yeah, I'd say we want the standard to be: every token with metadata will have a metadata pointer. That should always be the first step in discovery of said metadata